File: pinyougou-manager-web/src/main/java/com/pinyougou/manager/controller/BrandController.java
package com.pinyougou.manager.controller;

import java.util.List;
import java.util.Map;


import com.pinyougou.entity.PageResult;
import com.pinyougou.entity.Result;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import com.alibaba.dubbo.config.annotation.Reference;
import com.pinyougou.pojo.TbBrand;
import com.pinyougou.sellergoods.service.BrandService;

/**
 * 品牌控制层
 * @author a2363196581
 *
 */
@RestController
@RequestMapping("/brand")
public class BrandController {
	@Reference
	private BrandService brandService;

	@RequestMapping("/findAll")
	public List<TbBrand> findAll(){
		return brandService.findAll();
	}

	@RequestMapping("/findPage")
	public PageResult findPage(int page,int rows){
		return brandService.findPage(page,rows);
	}

	@RequestMapping("/add")
	public Result add(@RequestBody TbBrand tbBrand){
		try {
			brandService.add(tbBrand);
			return new Result(true,"添加成功");
		}catch (Exception e){
			return new Result(false,"添加失败");
		}

	}

	@RequestMapping("/update")
	public Result update(@RequestBody TbBrand tbBrand){
		try {
			brandService.update(tbBrand);
			return  new Result(true,"修改成功");
		}catch (Exception e){
			return new Result(false,"修改失败");
		}

	}

	@RequestMapping("/findOne")
	public TbBrand findOne(Long id){
		return brandService.findOne(id);
	}

	@RequestMapping("/dele")
	public Result dele(Long[] ids){
		try {
			brandService.dele(ids);
			return new Result(true,"删除成功");
		}catch (Exception e){
			return new Result(false,"删除失败");
		}
	}


	@RequestMapping("/search")
	public PageResult search(@RequestBody TbBrand tbBrand,int page,int rows){
		System.out.println(page);
		System.out.println(rows);
		return brandService.findPage(tbBrand,page,rows);
	}

	@RequestMapping("/findBrandList")
	public List<Map> findBrandList(){
		return brandService.selectOptionList();
	}
}

File: pinyougou-manager-web/src/main/webapp/js/controller/brandController.js
//定义控制器
app.controller('brandController', function($scope,$controller, $http, brandService) {

    $controller('baseController',{$scope:$scope});

    // 查询品牌列表
    $scope.findAll = function() {

        brandService.findAll().success(function(response) {
            $scope.list = response;

        });
    }

    // 分页查询
    $scope.findPage = function(page, rows) {
        brandService.findPage(page, rows).success(function(response) {
            $scope.list = response.rows;// 列表数据
            $scope.paginationConf.totalItems = response.total;// 总条数
        });
    }

    $scope.searchEntity = {};

    // 条件分页查询
    $scope.search = function(page, rows) {
        brandService.search(page, rows, $scope.searchEntity).success(
            function(response) {
                $scope.list = response.rows;// 列表数据
                $scope.paginationConf.totalItems = response.total;// 总条数
            });
    }



    // 查询品牌
    $scope.findOne = function(id) {
        brandService.findOne(id).success(function(response) {
            $scope.entity = response;
        });
    }

    // 新增品牌
    $scope.save = function() {
        var serviceObject;
        if ($scope.entity.id != null) {// 如果对象id不为空说明是修改操作
            serviceObject = brandService.update($scope.entity);
            alert(1);
        } else {
            serviceObject = brandService.add($scope.entity);
        }
        serviceObject.success(function(response) {
            if (response.success) {
                $scope.reloadList();
            } else {
                alert(response.message);
            }
        });
    }



    // 批量删除
    $scope.dele = function() {
        brandService.dele($scope.selectIds).success(
            function(response) {
            if (response.success) {
                $scope.reloadList();
                $scope.selectIds = [];
            } else {
                alert(response.message);
            }
        });
    }



});